File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using System.Collections;
using DevExpress.XtraEditors.Controls;

namespace WindowsApplication1
{

    public partial class Form1 : Form
    {
                private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) });
            return tbl;
        }
        

        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = CreateTable(20);
            gridView1.Columns[0].ColumnEdit = repositoryItemButtonEdit1;
        }

        int GetButtonEditValue()
        {
            return (int)spinEdit1.Value;
        }

        void UpdateButtonEdit()
        {
            EditorButton button = spinEdit1.Properties.Buttons[1];
            button.Enabled = !(spinEdit1.EditValue == null);
            button.Caption = rowVisibilityHelper1.IsRowInvisible(GetButtonEditValue()) ? "Show" : "Hide";

        }

        private void spinEdit1_TextChanged(object sender, EventArgs e)
        {
            UpdateButtonEdit();
        }

        private void spinEdit1_ButtonClick(object sender, ButtonPressedEventArgs e)
        {
            if (e.Button.Index == 1)
                rowVisibilityHelper1.ToggleRowVisibility(GetButtonEditValue());
            UpdateButtonEdit();
        }

        private void repositoryItemButtonEdit1_ButtonClick(object sender, ButtonPressedEventArgs e)
        {
            int focusedRowHandle = gridView1.FocusedRowHandle;
            rowVisibilityHelper1.HideRow(gridView1.GetDataSourceRowIndex(focusedRowHandle));
            gridView1.FocusedRowHandle = focusedRowHandle;
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports System.Collections
Imports DevExpress.XtraEditors.Controls

Namespace WindowsApplication1

	Partial Public Class Form1
		Inherits Form
				Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(DateTime))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) })
			Next i
			Return tbl
				End Function


		Public Sub New()
			InitializeComponent()
			gridControl1.DataSource = CreateTable(20)
			gridView1.Columns(0).ColumnEdit = repositoryItemButtonEdit1
		End Sub

		Private Function GetButtonEditValue() As Integer
			Return CInt(Fix(spinEdit1.Value))
		End Function

		Private Sub UpdateButtonEdit()
			Dim button As EditorButton = spinEdit1.Properties.Buttons(1)
			button.Enabled = Not(spinEdit1.EditValue Is Nothing)
			If rowVisibilityHelper1.IsRowInvisible(GetButtonEditValue()) Then
				button.Caption = "Show"
			Else
				button.Caption = "Hide"
			End If

		End Sub

		Private Sub spinEdit1_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles spinEdit1.TextChanged
			UpdateButtonEdit()
		End Sub

		Private Sub spinEdit1_ButtonClick(ByVal sender As Object, ByVal e As ButtonPressedEventArgs) Handles spinEdit1.ButtonClick
			If e.Button.Index = 1 Then
				rowVisibilityHelper1.ToggleRowVisibility(GetButtonEditValue())
			End If
			UpdateButtonEdit()
		End Sub

        Private Sub repositoryItemButtonEdit1_ButtonClick(ByVal sender As Object, ByVal e As ButtonPressedEventArgs) Handles repositoryItemButtonEdit1.ButtonClick
            Dim focusedRowHandle As Integer = gridView1.FocusedRowHandle
            rowVisibilityHelper1.HideRow(gridView1.GetDataSourceRowIndex(focusedRowHandle))
            gridView1.FocusedRowHandle = focusedRowHandle
        End Sub
    End Class
End Namespace